### PR TITLE
Adding multigpu test for cuda 11.6

### PR DESCRIPTION
**Pull request by atalman**
Adding multigpu test for cuda 11.6
As part of preparing CUDA 11.6 to be our stable CUDA version. Adding MultiGPU test for 11.6

### CODE REVIEW

**Review by weiwangmeta — APPROVED** [review]
LGTM

**Review by malfet — CHANGES_REQUESTED** [review]
Can you please remove 11.3 ones right away?

File: .github/workflows/periodic.yml [hunk 1]
```diff
@@ -78,33 +78,34 @@ jobs:
       AWS_OSSCI_METRICS_V2_ACCESS_KEY_ID: ${{ secrets.AWS_OSSCI_METRICS_V2_ACCESS_KEY_ID }}
       AWS_OSSCI_METRICS_V2_SECRET_ACCESS_KEY: ${{ secrets.AWS_OSSCI_METRICS_V2_SECRET_ACCESS_KEY }}
 
-  linux-bionic-cuda10_2-py3_9-gcc7-build:
-    name: linux-bionic-cuda10.2-py3.9-gcc7
+  linux-bionic-cuda11_6-py3_7-gcc7-debug-build:
```
Comment (huydhn) [inline .github/workflows/periodic.yml:81]: Any story behind why we have debug build for 3.7 here but not 3.9 like others

**Review by huydhn — APPROVED** [review]

File: .github/workflows/periodic.yml (hunk 1 above)
Comment (atalman) [reply .github/workflows/periodic.yml:81]: Hm, don't really know I think 1 debug build per python version is enough

Comment (malfet) [inline .github/workflows/periodic.yml:81]: Hmm, this does not seem to be related to PR in question, does it?

File: torch/testing/_internal/opinfo/definitions/linalg.py [hunk 2]
```diff
@@ -1276,6 +1276,9 @@ def make_input():
                 "test_fn_fwgrad_bwgrad",
                 dtypes=(torch.complex128,),
             ),
+            DecorateInfo(
+                toleranceOverride({torch.float64: tol(atol=1e-4, rtol=1e-4)})
+            ),
```
Comment (malfet) [inline torch/testing/_internal/opinfo/definitions/linalg.py:1281]: Hmm, this should nothing to do with multigpu, shouldn't it?
Also, would it be possible to limit it to just a particular test? I.e. using something along the following lines
```suggestion
            DecorateInfo(
                toleranceOverride({torch.float64: tol(atol=1e-4, rtol=1e-4)}),
                "TestGradients",
                "test_fn_gradgard",
                device_type="cuda",
            ),
```

Comment (Lezcano) [reply torch/testing/_internal/opinfo/definitions/linalg.py:1281]: Yes, this at first sight has nothing to do with multiGPU, but this is a test that is fairly flaky and it was activated last week, so I'm fine with running it with more lax tolerances.

**Review by lezcano — APPROVED** [review]
@malfet's review is on point. This should be ready to go once the more strict decorator is used.

**Comment by atalman** [comment]
Closing this one if favor of https://github.com/pytorch/pytorch/pull/84313
